src/main/kotlin/xyz/atrius/waystones/data/config/Localization.kt

```diff
@@ -0,0 +1,42 @@
+package xyz.atrius.waystones.data.config
+
+import org.bukkit.configuration.file.YamlConfiguration
+import xyz.atrius.waystones.utility.KotlinPlugin
+import xyz.atrius.waystones.utility.translateColors
+import java.io.File
+import java.io.InputStreamReader
+import java.text.MessageFormat
+import java.util.*
+
+class Localization(val plugin: KotlinPlugin, localeTag: String) {
+    private var configFile = File(plugin.dataFolder, "locale-$localeTag.yml")
+    private val config = YamlConfiguration()
+    private var locale: Locale = Locale.getDefault()
+
+    init {
+        reload(localeTag)
+    }
+
+    fun localize(key: String, vararg args: Any?): String {
+        return MessageFormat(config.getString(key)!!, locale)
```

Comment:
You should avoid using the `!!` operator since it can cause exceptions to be thrown if the key references a nonexistent value. In this case you should consider passing a default value using the `?:` operator.

Reply:
The `config.getString(key)` call should never return null since we applied in-jar defaults to the config. If it is null then the plugin is trying to read a non-existent string, which imo better not to fail silently.

Reply:
I can understand the rationale, but looking at it as an error using the `!!` a bit vague since it just returns a `NullPointerException`. Perhaps instead an error should be explicitly thrown instead? Example: ```kt
```kt
return MessageFormat(config.getString(key) ?: throw IllegalArgumentException("$key does not exist in configuration file!"), locale)
```